Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/safetynet/SafetyNetClientWrapper.kt

```diff
@@ -0,0 +1,116 @@
+package de.rki.coronawarnapp.datadonation.safetynet
+
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.common.api.CommonStatusCodes
+import com.google.android.gms.safetynet.SafetyNetApi
+import com.google.android.gms.safetynet.SafetyNetClient
+import com.google.gson.JsonObject
+import com.google.gson.JsonParser
+import dagger.Reusable
+import de.rki.coronawarnapp.datadonation.safetynet.SafetyNetException.Type
+import de.rki.coronawarnapp.environment.EnvironmentSetup
+import okio.ByteString.Companion.decodeBase64
+import timber.log.Timber
+import javax.inject.Inject
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+@Reusable
+class SafetyNetClientWrapper @Inject constructor(
+    private val safetyNetClient: SafetyNetClient,
+    private val environmentSetup: EnvironmentSetup
+) {
+
+    suspend fun attest(nonce: ByteArray): Report {
+        val jwsResult = callClient(nonce).jwsResult ?: throw SafetyNetException(
+            Type.ATTESTATION_FAILED,
+            "JWS was null"
+        )
+
+        val components = jwsResult.split(".")
+        if (components.size != 3) throw SafetyNetException(
+            Type.ATTESTATION_FAILED,
+            "Invalid JWS: Components are missing."
+        )
+
+        val header = try {
+            components[0].decodeBase64Json()
+        } catch (e: Exception) {
+            throw SafetyNetException(Type.ATTESTATION_FAILED, "Failed to decode JWS header.", e)
+        }
+        val body = try {
+            components[1].decodeBase64Json()
+        } catch (e: Exception) {
+            throw SafetyNetException(Type.ATTESTATION_FAILED, "Failed to decode JWS body.", e)
+        }
+
+        val signature = try {
+            components[2].decodeBase64()!!.toByteArray()
+        } catch (e: Exception) {
+            throw SafetyNetException(Type.ATTESTATION_FAILED, "Failed to decode JWS signature.", e)
+        }
+
+        return Report(
+            jwsResult = jwsResult,
+            header = header,
+            body = body,
+            signature = signature
+        )
+    }
+
+    private fun String.decodeBase64Json(): JsonObject {
+        val rawJson = decodeBase64()!!.string(Charsets.UTF_8)
+        return JsonParser.parseString(rawJson).asJsonObject
+    }
+
+    private suspend fun callClient(nonce: ByteArray): SafetyNetApi.AttestationResponse = suspendCoroutine { cont ->
+        safetyNetClient.attest(nonce, environmentSetup.safetyNetApiKey)
```

Comment:
A `NullPointerException` is thrown here when there is no `safetyNetApiKey` defined in the environment. The error message that the user/developer then sees is not really helpful. Maybe we could tell him something like "safety net API key missing, please add it in test_environments.json"?

Reply:
I updated the`test_environments` in our wiki so that it now includes these new keys.

Reply:
> A NullPointerException is thrown here when there is no safetyNetApiKey defined in the environment. The error message that the user/developer then sees is not really helpful. Maybe we could tell him something like "safety net API key missing, please add it in test_environments.json"?

I will add a more descriptive error to this and the other environment variables that can be accessed.

> I updated thetest_environments in our wiki so that it now includes these new keys.

I was waiting for the server environments to be final.

Reply:
As this has nothing to do with safetynet directly but our environments, I'd like to do this in an extra PR, would that be okay with you?

Reply:
Yes